Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/PropertyTraits.h

```diff
@@ -437,12 +437,12 @@ namespace ScriptCanvas
                         GetterFunction getterFunction = [&behaviorProperty, getterMethod](const Datum& thisDatum) -> AZ::Outcome<Datum, AZStd::string>
                         {
                             const size_t maxGetterArguments = 1;
-                            AZStd::array<AZ::BehaviorValueParameter, maxGetterArguments> getterParams;
+                            AZStd::array<AZ::BehaviorArgument, maxGetterArguments> getterParams;
                             const AZ::BehaviorParameter* thisParam = getterMethod->GetArgument(0);
-                            AZ::Outcome<AZ::BehaviorValueParameter, AZStd::string> thisObjectParam = thisDatum.ToBehaviorValueParameter(*thisParam);
+                            AZ::Outcome<AZ::BehaviorArgument, AZStd::string> thisObjectParam = thisDatum.ToBehaviorValueParameter(*thisParam);
```

Comment:
marking `thisObjectParam` -> `thisArg`

This case illustrates the issue with changing the C++ typename, but not changing the variable names.
Because `thisParam` is a BehaviorParameter, while the `thisObjectParam` is a BehaviorArgument.